src/anomalib/pipelines/components/actions/grid_search.py

```diff
@@ -0,0 +1,162 @@
+    grid_dict = {key: value for key, value in _container.items() if "grid" in key}
+    _container = {key: value for key, value in _container.items() if key not in grid_dict}
+    combinations = list(product(*convert_to_tuple(grid_dict.values())))
+    for combination in combinations:
+        for key, value in zip(grid_dict.keys(), combination, strict=True):
+            _container[key.removesuffix(".grid")] = value
+        yield to_nested_dict(_container)
+
+
+class GridSearchAction(ActionTypeHint):
+    """Grid search action for jsonargparse.
+
+    Allows using grid search key in the arguments.
+
+    Example:
+        ```yaml
+
+            nested_key:
+                class_path: str
+                init_args: [arg1, arg2]
+        ```
+        or
+        ```yaml
+            nested_key:
+                class_path:
+                    grid: [val1, val2]
+                init_args:
+                    - arg1
+                        grid: [val1, val2]
+                    - arg2
+                        ...
+        ```
+
+    """
+
+    def __call__(self, *args, **kwargs) -> "GridSearchAction | None":
+        """Parse arguments for grid search."""
+        if len(args) == 0:
+            kwargs["_typehint"] = self._typehint
+            kwargs["_enable_path"] = self._enable_path
+            return GridSearchAction(**kwargs)
+        return None
+
+    @staticmethod
+    def _crawl(value: Any, parents: list[str], parent: str = "") -> None:  # noqa: ANN401
+        """Crawl through the dictionary and store path to parents."""
+        if isinstance(value, dict):
+            for key, val in value.items():
+                if key == "grid":
+                    parents.append(parent)
+                elif isinstance(val, dict):
+                    parent_key = f"{parent}.{key}" if parent else key
+                    GridSearchAction._crawl(val, parents, parent_key)
+
+    @staticmethod
+    def _pop_nested_key(container: dict, key: str) -> None:
+        """Pop the nested key from the container."""
+        keys = key.split(".")
+        if len(keys) > 1:
+            GridSearchAction._pop_nested_key(container[keys[0]], ".".join(keys[1:]))
+        elif key != "":
+            container.pop(keys[0])
+        else:
+            # it is possible that the container is {"grid": [...]}
+            container.clear()
+
+    @staticmethod
+    def sanitize_value(value: dict | Namespace) -> dict:
+        """Returns a new value with all grid search keys removed."""
+        _value = dict_from_namespace(value) if isinstance(value, Namespace) else copy.deepcopy(value)
+        keys: list[str] = []
+        GridSearchAction._crawl(_value, keys)
+        for key in keys:
+            GridSearchAction._pop_nested_key(_value, key)
+        return _value
+
+    def _check_type(self, value: dict | Namespace | Any, append: bool = False, cfg: Namespace | None = None) -> Any:  # noqa: ANN401
+        """Ignore all grid search keys.
```

Comment:
Not sure if these docstrings are sphinx friendly. can you double check these. For example yaml examples could be done as follows: 

.. code-block:: yaml
    ....